crates/codegen/runtime/generator/src/ir/transformed.rs

```diff
@@ -0,0 +1,181 @@
+use codegen_language_definition::model;
+use indexmap::IndexMap;
+use serde::Serialize;
+
+use super::ir_model::{Choice, IrModel, Repeated, Separated, Sequence};
+
+#[derive(Default, Serialize)]
+pub struct TransformedIrModel {
+    pub source_name: String,
+
+    pub sequences: IndexMap<model::Identifier, TransformedSequence>,
+    pub choices: IndexMap<model::Identifier, TransformedChoice>,
+    pub repeated: IndexMap<model::Identifier, Repeated>,
+    pub separated: IndexMap<model::Identifier, Separated>,
+}
+
+#[derive(Clone, Serialize)]
+pub struct TransformedSequence {
+    pub fields: Vec<TransformedField>,
+    pub has_added_fields: bool,
```

Comment:
Out of curiosity, what's the reason to have fields being marked as removed, but top-level elements (Sequence, Choice, ...) directly removed from the maps?